src/Microsoft.DotNet.Interactive.PowerShell/AzureShellConnection.cs

```diff
@@ -118,6 +119,36 @@ internal class AzureTenantResponse
         public AzureTenant[] value { get; set; }
     }
 
+    internal class CloudShellSettings
+    {
+        public CloudShellSettingProperties properties { get; set; }
+    }
+
+    internal class CloudShellSettingProperties
+    {
+        public string preferredLocation { get; set; }
+        public StorageProfile storageProfile { get; set; }
+        public string preferredShellType { get; set; }
+    }
+
+    internal class StorageProfile
+    {
+        public string storageAccountResourceId { get; set; }
+        public string fileShareName { get; set; }
+        public int diskSizeInGB { get; set; }
+    }
+
+    internal class FailedRequest
+    {
+        public FailedRequestError error { get; set; }
+    }
+
+    internal class FailedRequestError
+    {
+        public string code { get; set; }
+        public string message { get; set; }
+    }
```

Comment:
Doesn't Newtonsoft let you use capital letters?

Reply:
Yes but it looks here like serialization is being handled by System.Text.Json, which is fine. We'll eventually move everything to that serializer in any case. Because these classes are internal and purely for serialization, this approach is arguably clearer than following C# naming conventions.

Reply:
Eventually, we want to replace all the HTTP authentication code with the C# Azure autorest library, though that work is still in backlog.